perl-lib/OESS/lib/OESS/Database.pm

```diff
@@ -10197,6 +10219,72 @@ sub revert_edge_interface_move_maintenance {
     };
 }
 
+=head2 move_vrf_endpoints
+
+=cut
+
+sub move_vrf_endpoints {
+    my ($self, %args) = @_;
+    my $orig_interface_id  = $args{'orig_interface_id'};
+    my $new_interface_id   = $args{'new_interface_id'};
+    my %used_vlans;
+    
+    # Gather occupied vlans from circuits on the new interface
+    my $dst_edge_interface_recs = $self->get_circuit_edge_interface_memberships(
+        interface_id => $new_interface_id);
+    foreach my $edge_int_rec (@$dst_edge_interface_recs) {
+        $used_vlans{$edge_int_rec->{'external_vlan_id'}} = 1;
+    }
+
+    # Gather occupied vlans from vrf endpoints on the new interface
+    my $query = "SELECT tag FROM vrf_ep ".
```

Comment:
Can we not use a library call instead?  It might be interesting to be able to do something like OESS::DB::Endpoints::get_endpoints(interface_id => $interface_id) and get back an array of OESS::Endpoint objects instead of doing the 2 individual queries... (better for reuse later)

Reply:
Yep, will do this.